### PR TITLE
Powertube

### PR DESCRIPTION
**Pull request by Sylvaritius**
Andet forsøg på en pull request, denne gang uden merge conflics. Venligst review.
m.v.h
Mads Christensen

### CODE REVIEW

File: Microwave.Classes/Controllers/UserInterface.cs [hunk 1]
```diff
@@ -42,6 +44,14 @@ private enum States
             myCooker = cooker;
             myLight = light;
             myDisplay = display;
+            if (maxPower < 50)
+            {
+                throw new ArgumentOutOfRangeException("Maxpower", _maxPower, "Must be at least 50");
```
Comment (CodeOink) [inline Microwave.Classes/Controllers/UserInterface.cs:49]: Fin måde også, at kunne teste senere på en exception

File: Microwave.Test.Unit/PowerTubeTest.cs [hunk 2]
```diff
@@ -61,5 +62,83 @@ public void TurnOn_WasOn_ThrowsException()
             uut.TurnOn(50);
             Assert.Throws<System.ApplicationException>(() => uut.TurnOn(60));
         }
+
+        [TestCase(-1)]
+        [TestCase(0)]
+        [TestCase(49)]
```
Comment (CodeOink) [inline Microwave.Test.Unit/PowerTubeTest.cs:68]: Meget fedt med 3 test Cases her, som alle giver det samme. Godt du har udvalgt meget specifikt de scenarier, som giver edge cases.

File: Microwave.Test.Unit/PowerTubeTest.cs [hunk 3]
```diff
@@ -61,5 +62,83 @@ public void TurnOn_WasOn_ThrowsException()
...
+        public void CreateNewPowerTube_ThrowsException(int maxPower)
+        {
+
+            Assert.Throws<System.ArgumentOutOfRangeException>(() => new PowerTube(output, maxPower));
+        }
+       
+        [TestCase(50)]
+        [TestCase(500)]
+        [TestCase(5000)]
```
Comment (CodeOink) [inline Microwave.Test.Unit/PowerTubeTest.cs:77]: godt

File: Microwave.Classes/Controllers/UserInterface.cs [hunk 4]
```diff
@@ -60,7 +70,7 @@ public void OnPowerPressed(object sender, EventArgs e)
                     myState = States.SETPOWER;
                     break;
                 case States.SETPOWER:
-                    powerLevel = (powerLevel >= 700 ? 50 : powerLevel+50);
+                    powerLevel = (powerLevel >= _maxPower ? 50 : powerLevel+50);
```
Comment (CodeOink) [inline Microwave.Classes/Controllers/UserInterface.cs:73]: fedt lille ekstra tjek

**Review by CodeOink — APPROVED** [review]
Fint, og gode stykker. Ikke Noget ellers at tilføje. Virker som noget fornuftigt kode.
